agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -114,10 +115,18 @@ public class OpenTelemetry {
             pipeline.interceptAgentRunError(interceptContext) { eventContext ->
                 logger.debug { "Execute OpenTelemetry agent run error handler" }
 
+                val errorStatus = SpanEndStatus(code = StatusCode.ERROR, description = eventContext.throwable.message)
+
+                val events = buildList {
+                    add(ExceptionEvent(throwable = eventContext.throwable))
+                }
+
                 // Make sure all spans inside InvokeAgentSpan are finished
                 spanProcessor.endUnfinishedInvokeAgentSpans(
                     agentId = eventContext.agentId,
-                    runId = eventContext.runId
+                    runId = eventContext.runId,
+                    spanEndStatus = errorStatus,
```

Comment:
major: We should not set common spans or common status for all messages when finish unstopped spans. This is an "internal system" method that should not handle such cases. We rather need to handle this inside the framework and provide you with necessary events and attributes to update/set the status for a particular span.